UTC-179: Bythenumber improvements

brittrich opened the pull request:
Testing integration of simplified UTCWidgets module to streamline field use.

Commit:
UTC-179: Adding fontawesome icon loop.

brittrich commented on apps/drupal-default/particle_theme/templates/block/block--utc-highlights-block--default.html.twig:
```diff
@@ -111,42 +87,15 @@
 					<hr class="border-t-3 border-utc-new-gold-500 mt-3 mb-6 ml-8 w-20" />
 				{% endif %}
 					<ul class="flex flex-row flex-wrap list-none pl-0 mb-0">
```
@bmartinez287 

The loop below is very similar to how I did this in the button group...thinking accessing variables might be similar to the "Link" field type. However, this isn't working for me. I tried a few different variations and then backed up from the loop to explore drilling down to the items without a loop. No dice so far after a good bit of research and testing. When I started getting in to the module files/php to determine if something needed reworking there I figured I should check with you first in case the answer was an obvious one! Very possible. ;)

So...any ideas?

If not, I can keep researching.

If we can get them to render individually, I think the rewrite to use the widget here will be pretty straightforward from there.

brittrich replied:
Oh, and to be clear, "field_highlighted_icon" is the name of the UTCWidget field.

bmartinez287 replied:
`  {% set card_icon = content.field_bas|field_value %}`
 ` {% set card_icondescription = card_icon.0['#icons'][0]['#description'] %}`
 ` {% set card_iconinfo = card_icon.0['#icons'][0]['#info'] %}`
`  {% set card_iconname = card_icon.0['#icons'][0]['#name'] %}`

That;s what you were looking for. The only thing is you need to iterate with a for loop on the second 0
['#icons'][0]['#description']  that one.

bmartinez287 replied:
Because of the for loop i probably would not leave those variables on the top of the page as variables but i used that for debugging. Also field_bas is your field_highlighted_icon.
Lastly because we are getting the information in such a granular form.
<img width="586" alt="Screen Shot 2021-10-18 at 8 38 25 PM" src="https://user-images.githubusercontent.com/39039024/137825075-76b0d6ee-e47f-4e3b-93cb-6b9e70fad21a.png">

brittrich replied:
Woo hoo! Thanks, Bernardo. If we have time sometime, I'd love to meet up and talk through the "why/how" of this. Kint wasn't working well for me for some reason this go around (no formatting...just a massive dump of info...even with the module on). And, even when kint is working, I know there are better ways to get the variables that I likely just don't have set up (xdebug).

bmartinez287 replied:
yeah, I can show on Thursday at 4 pm if that time works for you let me know.

Commit:
UTC-179: Start to updating variables.

Commit:
UTC-179: Update loop to work.

Commit:
UTC-179: Second loop variable adjustmentsk.

Commit:
UTC-179: Altering fields and combinging templatesk.

Commit:
UTC-179: Remove extra rows field.

brittrich commented on apps/drupal-default/particle_theme/templates/block/block--utc-highlights-block--default.html.twig:
```diff
@@ -71,160 +52,87 @@
+					{% for content in info_1.0['#icons'] if info_1.0['#icons'] is not empty %}
+					<li class="p-10 text-xl text-center xl:w-1/2 w-full {{description_color }}">
+						{% if info_1.0['#icons'][loop.index0]['#name'] %}
+						<div class="mb-4 {{ icon_color }}"><span class="fas fa-2x {{ info_1.0['#icons'][loop.index0]['#name'] }}" ></span></div>
```
Accessing the font awesome variable like this would allow us to force a size/style on the icons instead of the widget field providing those options for the user to select (FA additional style options). I think we'd probably prefer this way for consistency, but lmk if I'm wrong!

bmartinez287 replied:
I think that's great.

brittrich replied:
Do you think it will be confusing if the additional options are still there in the editor interface (but don't do anything)? I could add a note.

brittrich commented:
@bmartinez287 and @UTCbridget I think the UTC modified FA field widget will work for the By the Numbers / Highlighted Info block. Having less fields to access for content made the template much simpler, so I combined the two view modes (side box and full) in to a single template using a select field. I just need to finish up with some field deletion/renaming and then I want to run a few more tests before marking this ready. 

Lando keeps shutting down (almost like it doesn't like it when I delete a field?) and now I can't get Docker going again, it just waits forever....I'll troubleshoot tomorrow and try to get a gif/example added to this PR.

bmartinez287 commented:
That's odd, try doing a lando drush cr. It might be that twig is not hot reloading the cache. I will take a look this week.
Whenever you see that try a 
```
lando drush cr 
lando start
```

brittrich commented:
@bmartinez287 I think I found an issue with the custom widget on some further testing. So, I deleted and re-added the UTC Widget fields so I could rename them. Afterward, they show in the "disabled" area for the fields, but there is an error when I move the field to the visible area and save (see below). Also, the dropdown for display options is empty for the field.

What's interesting, though, is that the fields were working in the block and template before I deleted them to re-add...but I don't think I had tried to reposition them in the field form display at all, so perhaps this error was there all along I just didn't find it until now.  

Here are a few screenshots:
<img width="807" alt="Screen Shot 2021-10-26 at 11 37 07 PM" src="https://user-images.githubusercontent.com/50490141/138996248-869ca974-ab3a-46a6-bcd7-22184dfcd152.png">


<img width="1764" alt="Screen Shot 2021-10-26 at 11 35 19 PM" src="https://user-images.githubusercontent.com/50490141/138996259-33050311-f3d6-4ead-ad55-f03141f7891d.png">


If you have any troubleshooting suggestions, happy to try them out. So far, I've tested without the field at all, verified no error so I think the error is tied to the field.

bmartinez287 commented:
Try lando drush cr
After that refresh the page if that does not do it let me know.
I seen something similar but that’s usually because Memcache is in the
local environment abs it can step on it.

On Tue, Oct 26, 2021 at 11:50 PM brittrich ***@***.***> wrote:

> @bmartinez287 <https://github.com/bmartinez287> I think I found an issue
> with the custom widget on some further testing. So, I deleted and re-added
> the UTC Widget fields so I could rename them. Afterward, they show in the
> "disabled" area for the fields, but there is an error when I move the field
> to the visible area and save (see below). Also, the dropdown for display
> options is empty for the field.
>
> What's interesting, though, is that the fields were working in the block
> and template before I deleted them to re-add...but I don't think I had
> tried to reposition them in the field form display at all, so perhaps this
> error was there all along I just didn't find it until now.
>
> Here are a few screenshots:
> [image: Screen Shot 2021-10-26 at 11 37 07 PM]
> <https://user-images.githubusercontent.com/50490141/138996248-869ca974-ab3a-46a6-bcd7-22184dfcd152.png>
>
> [image: Screen Shot 2021-10-26 at 11 35 19 PM]
> <https://user-images.githubusercontent.com/50490141/138996259-33050311-f3d6-4ead-ad55-f03141f7891d.png>
>
> If you have any troubleshooting suggestions, happy to try them out. So
> far, I've tested without the field at all, verified no error so I think the
> error is tied to the field.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/UTCWeb/particle/pull/347#issuecomment-952512826>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJJ3AMA5RBVFEMPI6B7HYTDUI5ZHNANCNFSM5GFP4KJA>
> .
>

Commit:
UTC-179: some minor field renaming

bmartinez287 commented:
@brittrich, yeah it seems like I'm missing something in there. If you could circle back to this issue tomorrow, that be great. I will try to have it fine tune by the end of the day. Else, I will message you once is ready.

bmartinez287 commented:
I figured it out. I made a noobie mistake because I refactored the annotation id on the field but forgot to update it elsewhere(widget and formatter). I will push the PR today. After you pull upstream develop from utccloud and do a lando drush cr you should be good to go. @brittrich

brittrich commented:
@bmartinez287, a few observations after testing a bit more on this new widget/config:

1. The description field limit looks like it is 280 characters. Seems reasonable, but should we add a note to the description field that this is the case?

2. If we could remove the "Additional" options that don't actually apply, that would be great, I think, and save a lot of space/confusion.

3. If I leave the icon blank, but have a title and description, it won't show anything and won't keep what I enter in the title/description field without an icon. Previously, I had the icon as optional in case there was a need for a simple number w/ text, but it looks like the widget field may require it. I can't try out looping generically or on something like info/title because the input isn't kept for title and description. Right now, it is looping on whether an icon is present.

**User Interface:**

![examples-highlight-info-4](https://user-images.githubusercontent.com/50490141/139620940-c7b9d273-2fc4-4423-80f0-4e59e9ba7a21.gif)

brittrich commented:
@bmartinez287 and @UTCbridget here are some examples of where this is now:

![examples-highlight-info-5](https://user-images.githubusercontent.com/50490141/139620991-1048acca-d12e-43aa-8371-55230e29a42f.gif)



![examples-highlight-info-3](https://user-images.githubusercontent.com/50490141/139621039-a6ee27fb-01b9-4e33-9b34-4bd28c91ed48.gif)

bmartinez287 commented:
1.  Yeah, 200 might be too long. I was trying to keep it away from being a body field without CKeditor.  Mostly because that could encourage large paragraphs. We could add a description that's a great idea. It's 200, at the database layer.
2. Yes let's try doing that
3. That's because I extended the code for the original fontawesome but did not alter its requirements. For this type of block, it would make sense to update that. Make it so none of the fields are required at the database or PHP layer but they are required at the twig level.  That way we could have only an icon or only an info field.

bmartinez287 commented:
@brittrich, Im close on this one.
1.  Allowing any field to be the primary field as you asked in question 3 above its fairly easy i got that already.
2.  Hiding that additional settings option is trickier. Mostly because the fontawesome module makes a few assumptions that the data is going to be there and given to it, even if is just its default.
3. We could add the description field note the same way the inco-fa has one. What do you think we should have there?

bmartinez287 commented:
be sure to do a git pull upstream develop on utccloud to get those changes and then refresh the page.

bmartinez287 commented:
How did you know the limit was 280? Maybe I need to fix that since the database field has a limit of 200. @brittrich

bmartinez287 commented:
@brittrich I got it all working check it out and let me know/

brittrich commented:
> How did you know the limiit was 280? Maybe i need fix that, since the database field has a limit of 200. @brittrich

Wasn't an exact science, I typed until it stopped and then put it in a character counter. :)

brittrich commented:
Looks great! Everything appears to be working on this now so I'm marking it ready for review.

<img width="1015" alt="Screen Shot 2021-11-02 at 11 36 27 PM" src="https://user-images.githubusercontent.com/50490141/140006832-9c201696-83fe-449a-90ce-86af86f64535.png">

Commit:
UTC-179: rename and  delete files to remove unused view modes.

bmartinez287 commented:
```
#size' => 60,
'#maxlength' => 128
```
I likely swap size for maxlenght on the fields.
https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21Element%21Textfield.php/class/Textfield/9.3.x
I not sure size serves the purpose that drupal has for it as it does not change the size of the input field.

bmartinez287 commented:
We might be missing some configuration. I don't have some items like the text on top of section 1. Maybe we need to to be pushed to utccloud. @brittrich

UTCbridget commented:
You've done a fantastic job with this, Brittany! I can't wait to use it. I pull your branch to my local just for review. Here are some things I was thinking:

**CMS:**

1. I think "Info Title" and "Info" and "Info"may not be clear enough. What about "Highlight Title" or "Block Title" for "Info Title" and "Stat Headline" for the "Info" field? If we use "Stat Headline", maybe we also use "Stat Description"?
2. I used the maxLength for the Image Hovers. It has a nice countdown of characters for each field that uses it. Maybe we do this here? @bmartinez287, are you working on this?
It's found under the Manage Form Display:
![Screen Shot 2021-11-03 at 4 34 14 PM](https://user-images.githubusercontent.com/82905787/140188612-eb092cc1-b70e-457f-b889-4ded45d964f6.png)

**Frontend Suggestions:**

1. With regards to the side box, I'm think it should look a little different from the hero. I filled out the description boxes to the max and there was a large margin at the bottom of it that some people might think is a mistake, idk. Maybe contain it and center vertically against the background? I did some tinkering locally:

Before: 
![Screen Shot 2021-11-03 at 2 02 02 PM](https://user-images.githubusercontent.com/82905787/140189254-f16e81f1-7358-42de-a89a-4dabc4b54901.png)

After:
![Screen Shot 2021-11-03 at 4 18 58 PM](https://user-images.githubusercontent.com/82905787/140188985-0cb9ceed-d3a0-49a4-a8f1-f8dd8370e1ad.png)

2. Also, maybe we could make the icons larger (3xl), and the Info headline smaller (text-3xl)?

I've got the code already for the frontend changes if you're okay with it.

brittrich commented:
> You've done a fantastic job with this, Brittany! I can't wait to use it. I pull your branch to my local just for review. Here are some things I was thinking:
> 
> **CMS:**
> 
> 1. I think "Info Title" and "Info" and "Info"may not be clear enough. What about "Highlight Title" or "Block Title" for "Info Title" and "Stat Headline" for the "Info" field? If we use "Stat Headline", maybe we also use "Stat Description"?
> 2. I used the maxLength for the Image Hovers. It has a nice countdown of characters for each field that uses it. Maybe we do this here? @bmartinez287, are you working on this?
>    It's found under the Manage Form Display:
>    ![Screen Shot 2021-11-03 at 4 34 14 PM](https://user-images.githubusercontent.com/82905787/140188612-eb092cc1-b70e-457f-b889-4ded45d964f6.png)
> 
> **Frontend Suggestions:**
> 
> 1. With regards to the side box, I'm think it should look a little different from the hero. I filled out the description boxes to the max and there was a large margin at the bottom of it that some people might think is a mistake, idk. Maybe contain it and center vertically against the background? I did some tinkering locally:
> 
> Before: ![Screen Shot 2021-11-03 at 2 02 02 PM](https://user-images.githubusercontent.com/82905787/140189254-f16e81f1-7358-42de-a89a-4dabc4b54901.png)
> 
> After: ![Screen Shot 2021-11-03 at 4 18 58 PM](https://user-images.githubusercontent.com/82905787/140188985-0cb9ceed-d3a0-49a4-a8f1-f8dd8370e1ad.png)
> 
> 1. Also, maybe we could make the icons larger (3xl), and the Info headline smaller (text-3xl)?
> 
> I've got the code already for the frontend changes if you're okay with it.

@UTCbridget YES, these design changes look great to me! I like the box contained. If you've already got them ready, please feel free to add them!

How about "Item Headline" and "Item Description" and "Item Icon"? Just thinking on occasion someone might put text instead of a stat as the headline. Also, instead of Highlighted Info 1 how about the overall field being called "Highlighted Items 1" and "Highlighted Items 2". What do you think?

I'll let @bmartinez287 chime in on the textfield option. I'm not exactly sure how we would integrate that on this text field since it is part of the new widget...but maybe it's possible!

bmartinez287 commented:
I was looking into that text field option. That's actually harder than it seems as this has a custom widget. I tried adding a patch to the maxlenght module but I got it 50% there not fully working yet.
@brittrich is there anything we could add to the solid blue or solid gray rectangle background to improve it.
A solid color seems fine but it seems like we could add something there.

bmartinez287 commented:
![utkhomepage](https://user-images.githubusercontent.com/39039024/140245146-3a7d9cc1-a517-4c4c-95f8-7e895db9091d.PNG)
![utaustin](https://user-images.githubusercontent.com/39039024/140245152-e1a70293-168e-4b16-aa94-f38a83f2793e.PNG)
![ataustin](https://user-images.githubusercontent.com/39039024/140245155-f95b4695-1e29-4334-a0bf-3962fec4e0d3.PNG)
https://www.utexas.edu/
https://utk.edu/
I was thinking something like that. I know, we got spoiled with the hero backgrounds.

bmartinez287 commented:
That last one is super subtle but it adds a nice touch to it. @brittrich

bmartinez287 commented:
I will stop by the graphic designer's desks tomorrow to see if they have something that could easily be placed there. Or some awesome idea for it. Maybe they have something that could be repurposed for it but if you have some other ideas that be great too.

UTCbridget commented:
@bmartinez287 I love the idea of different graphical backgrounds.

UTCbridget commented:
@brittrich, your "Item" suggestions work perfectly for me! Much better than "Stat"

UTCbridget commented:
@brittrich and yes, too, on the "Highlighted Items 1" and "Highlighted Items 2". That offers more clarity for the user.

UTCbridget committed:
UTC-179: Remove the 43 extra files.

brittrich commented:
> @brittrich and yes, too, on the "Highlighted Items 1" and "Highlighted Items 2". That offers more clarity for the user.

@bmartinez287 let's go with these in the widget, then: "Item Headline" and "Item Description" and "Item Icon".

I'll fix the header fields to "Highlighted Items 1 and 2."

brittrich committed:
Merge pull request #5 from UTCbridget/brittrich-feature/UTC-179-BytheNumbersBlock-improvements

Brittrich feature/utc 179 bythe numbers block improvements

brittrich commented:
I can look some more (unless designers or Bridget have any ideas), but here are a few options I was playing around with:

Textures:
<img width="1584" alt="Screen Shot 2021-11-04 at 10 34 50 PM" src="https://user-images.githubusercontent.com/50490141/140453664-97e6fbf7-bff5-4794-a03a-77a66cf34311.png">
<img width="1584" alt="Screen Shot 2021-11-04 at 10 32 40 PM" src="https://user-images.githubusercontent.com/50490141/140453674-401f74d5-415e-43c3-816b-11270741f8b1.png">

Two shades:

<img width="1566" alt="Screen Shot 2021-11-04 at 11 26 55 PM" src="https://user-images.githubusercontent.com/50490141/140453689-1d0bb0b4-23d6-41ee-adc3-61967e0859c4.png">

There could be another texture in that one, or perhaps a third shade.

brittrich commented:
Okay, here are a few options using what marketing sent and a few I found as well. Which do you all prefer? The above samples are still options, they would just be integrated in a different way than the below since they are designed to be transparent texture patterns for css. Below are images with a transparent shadow overlay that allows them to be colored.

**Feather 1 (looks a bit like cardboard to me?)**
<img width="1634" alt="Screen Shot 2021-11-08 at 10 33 45 AM" src="https://user-images.githubusercontent.com/50490141/140777461-1b9fa3ef-5f0a-4c9a-8360-f79403d7a470.png">


<img width="1247" alt="Screen Shot 2021-11-08 at 10 33 33 AM" src="https://user-images.githubusercontent.com/50490141/140777468-d0c46ed0-f04e-40d4-8b7c-f15dcd8e24b9.png">


**Feather 2**

<img width="1635" alt="Screen Shot 2021-11-08 at 10 37 12 AM" src="https://user-images.githubusercontent.com/50490141/140777510-96365856-977f-4159-a64f-f8b6285ee62d.png">

<img width="1250" alt="Screen Shot 2021-11-08 at 10 37 02 AM" src="https://user-images.githubusercontent.com/50490141/140777556-eb160a33-f00b-44fe-9822-d67d4c13dec2.png">

**Crinkled Paper (I found this one -- the marketing paper one was a bit too subtle)**

<img width="1606" alt="Screen Shot 2021-11-08 at 11 10 32 AM" src="https://user-images.githubusercontent.com/50490141/140777622-88e795a1-0a52-4946-9cbc-5141d1304f45.png">

<img width="1214" alt="Screen Shot 2021-11-08 at 11 10 22 AM" src="https://user-images.githubusercontent.com/50490141/140777667-ac7f5fb0-f799-40ab-b593-c41e4464af98.png">

**Concrete (I added)**

<img width="1598" alt="Screen Shot 2021-11-08 at 11 05 28 AM" src="https://user-images.githubusercontent.com/50490141/140777755-4fbbd8d1-7ded-40b2-ad8a-710b8c1fe298.png">


<img width="1214" alt="Screen Shot 2021-11-08 at 11 05 43 AM" src="https://user-images.githubusercontent.com/50490141/140778126-cbb42cc3-7693-4093-b26b-62d16a9f80f1.png">

**Interesting Accident -- transparent so that the base image shows through a bit**

<img width="1598" alt="Screen Shot 2021-11-08 at 11 01 52 AM" src="https://user-images.githubusercontent.com/50490141/140777846-3f6de5c5-eeb7-4a9d-b305-3a9444d3f177.png">


<img width="1243" alt="Screen Shot 2021-11-08 at 10 58 19 AM" src="https://user-images.githubusercontent.com/50490141/140777872-69381c17-a8d5-4f74-9ad1-70a504c03803.png">

bmartinez287 commented:
Number 1 from the previous batch and number 1 from this batch have an interesting eye catching allure. Bonus points that we are using something they gave us. Maybe that might inspire them to share more items.

UTCbridget commented:
Wow. Those turned out much better than I thought they would! (And I agree about the cardboard.)

UTCbridget commented:
My votes:

1. the first speckled background
2. the cardboard background, but with less contrast so as not to compete so much with the type
3. the transparent backgrounds, but change the gray transparent to a white transparent, maybe?

UTCbridget commented:
4. the diagonal section

bmartinez287 commented:
![Screen Shot 2021-11-08 at 12 51 21 PM](https://user-images.githubusercontent.com/39039024/140792852-1fbd345f-2f91-4865-8e0d-70ac56f30126.png)
So I was showing Chris the new component and he had some interesting insights. He asked if adding links was possible.
I hesitated to say yes because I don't want this component to be a replacement to the buttons block and I'm pretty sure if every icon can have a link this is where that will go. However, some people might want to redirect those interested into another page with more details. Hence it might be good to add a link field type to the block itself. Only one, a global one that could provide such option. @brittrich

bmartinez287 commented:
I was playing around with button placement options on the previous pic. But it could be something else.

bmartinez287 commented:
Actually now I think about it i made those buttons to close to the edge on the pic above which could be problematic for full-width layouts.

britt-rich committed:
UTC-179: add in bg image for side box and link examplep.

brittrich commented:
Sounds like Number 1 from marketing is the winner! I adjusted like Bridget suggested with a darker overlay, good idea!

Also, @bmartinez287 here's one way we could do the button option, what do ya'll think? I don't have a field for it yet, but this is an example hard coded in the template. If we want something like this I can get a field added as well.

<img width="1785" alt="Screen Shot 2021-11-09 at 12 10 40 AM" src="https://user-images.githubusercontent.com/50490141/140866349-b4022712-d7ef-4a19-8dbb-7fd86742b3f7.png">


<img width="1785" alt="Screen Shot 2021-11-09 at 12 10 29 AM" src="https://user-images.githubusercontent.com/50490141/140866364-30e08c89-6d37-413e-b86f-28d827b7fd5e.png">
<img width="1218" alt="Screen Shot 2021-11-09 at 12 09 24 AM" src="https://user-images.githubusercontent.com/50490141/140866369-2a80f778-fca5-4153-b5d7-76b9a2035026.png">

bmartinez287 commented:
Actually, I forgot to add here that I talked to Chris and Lacy(graphic designer) from marketing and they leaned a bit more for this one. https://user-images.githubusercontent.com/50490141/140453664-97e6fbf7-bff5-4794-a03a-77a66cf34311.png

bmartinez287 commented:
![Screen Shot 2021-11-09 at 1 43 25 PM](https://user-images.githubusercontent.com/39039024/140985291-b689e052-6662-46f4-ad3a-8a40e34f930c.png)
If there's no icons could we increase the size of the font size in yellow? t seems very small if you only need text.

bmartinez287 commented on apps/drupal-default/particle_theme/templates/block/block--utc-highlights-block.html.twig:
```diff
@@ -0,0 +1,164 @@
+        <ul class="flex flex-row flex-wrap list-none pl-0 mb-0">
+          {% for content in info_1.0['#icons'] if info_1.0['#icons'] is not empty %}
+            <li class="p-10 text-lg text-center xl:w-1/2 w-full {{ description_color }}">
+              {% if info_1.0['#icons'][loop.index0]['#name'] %}
```
It probably be adding to this, as an if else to check on whether there's none then higher font size.

brittrich replied:
Added!

britt-rich committed:
UTC-179: add button fields and update styling to white outline box.

britt-rich committed:
UTC-179: Update item text size if icon is not present.

britt-rich committed:
UTC-179: Fix if statement for icons to remove html if no icon is present.

britt-rich committed:
UTC-179: Update BG images to transparent speckled asfalt.

britt-rich committed:
UTC-179:Update button hover interaction .

brittrich commented:
I added a few new edits:

1. Bigger font size for headline if no icon.
2. Adjustment to link style and hover interaction. I'd like to know what you think on the center alignment and the lack of border around the link. I debated a white border around (like hero button) and aligning the link to the right.
3. Updated bg for side box to speckled version. Used a transparent that had options for both light/dark. Think it has a bit more "speckle" than the first one I used.

<img width="1588" alt="Screen Shot 2021-11-10 at 11 37 22 PM" src="https://user-images.githubusercontent.com/50490141/141238392-7652466d-5d4b-44ed-a055-0f90fed9dc12.png">

![examples-highlight-info-6](https://user-images.githubusercontent.com/50490141/141238615-a936ab1e-34a9-48d8-b35f-34c881c82354.gif)

Fields that limit characters look like they are working well, thanks Bernardo!

<img width="1011" alt="Screen Shot 2021-11-10 at 11 24 16 PM" src="https://user-images.githubusercontent.com/50490141/141238659-e4820fc0-1fd3-4a84-8453-2456cb324865.png">

bmartinez287 commented:
this looks great!!! Anything else left? else i likely test it quickly and merge into particle. @brittrich @UTCbridget

brittrich commented:
@bmartinez287 I think it's ready to test and merge! At least, there's nothing else pending on my end. If you see any issues on testing just holler!
